### DISCUSSION TITLE
Upload Attachment via public API (Problem with payload format)

### DISCUSSION THREAD

**Discussion by julianmenav** [Budibase/budibase#10286] [Help]
I am trying to upload images to budibase storing as default in Minio. The app is self hosted.

Im sending a post request to the url: https://budibase.app/api/attachments/${bbtableid}/upload
with of course all headers already set, but dont really know how the payload has to be.

I have already seen this disscussion : 
https://github.com/Budibase/budibase/discussions/3725 and try to replicate it in my python app in which I think the payload has the same format : 

```
url = f"{budibase_url}/api/attachments/{table_id}/upload"
headers = {
    'x-budibase-app-id': bbappid,
    'x-budibase-api-key': bbapikey
} 
form = [('file', (open('logo.png', 'rb')))]
response = request.post(url, headers=headers, files=form)
```

It always respond with the error message 500:"Cannot read properties of undefined (reading 'file')"

All other API calls like retrieving rows, tables or update them works perfectly but with this one im struggling.

It would be very much appreciated to know how to upload an image as attachment to budibase from my python code, or how the payload of the request should be. Thanks in advance

**Comment by melohagan** [comment]
Hey @Julianmenav 

Looking at the code, it appears that file is expected as a **File** object with a number of properties including a write stream.

![Screenshot 2023-04-13 at 14 18 44](https://user-images.githubusercontent.com/101575380/231771016-31010231-7c0e-432b-aff6-15406d95f04e.png)

I'm not sure how you would convert between the JavaScript **FormData** object and a python type. It's worth noting that this endpoint is an internal one, and is designed to be used by the **AttachmentField** in the UI. _It is not available via the public API._

**Reply by julianmenav** [reply]
Thank you for your answer @melohagan .
Besides it seems to be an internal endpoint, in this discussion https://github.com/Budibase/budibase/discussions/3725 , the user @xnohat used it in a nodejs script so I think it can also be used via public API.
